### PR TITLE
[Fix] Fix the unit tests in C# 

### PR DESCRIPTION
**Pull request by vincentpierre**
Since the academy now resets in the first fixed update and not the awake method in inference, the episode counting was off in the Unity unit tests

### CODE REVIEW

File: unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs [hunk 1]
```diff
@@ -459,27 +489,28 @@ public void TestAcademy()
...
                 stepsSinceReset += 1;
-                if ((i % 20 == 0) && (i>0))
+                if ((i % 20 == 0) )
                 {
                     numberReset += 1;
                     stepsSinceReset = 1;
 
                 }
                 AcademyStepMethod.Invoke((object)aca, new object[] { });
+
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs]: remove empty lines

File: unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs [hunk 2]
```diff
@@ -240,31 +242,42 @@ public void TestAgent()
 
             AgentEnableMethod.Invoke(agent1, new object[] { aca });
             AcademyInitializeMethod.Invoke(aca, new object[] { });
-            AgentEnableMethod.Invoke(agent2, new object[] { aca });
 
             MethodInfo AcademyStepMethod = typeof(Academy).GetMethod("_AcademyStep",
                            BindingFlags.Instance | BindingFlags.NonPublic);
 
+            int numberAgent1Reset = 0;
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs]: ensure that you convert tabs to spaces, indentation starting from lines 243 seems off.

File: unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs [hunk 3]
```diff
@@ -240,31 +242,42 @@ public void TestAgent()
...
+                {
+                    numberAgent1Reset += 1;
+                }
+                if (i == 2) //Agent 2 is only initialized at step 2
+                {
+                    AgentEnableMethod.Invoke(agent2, new object[] { aca });
+                    numberAgent2Initialization += 1;
+                }
+
+                if ((i % 3 == 0) && (i > 2))
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs]: comment this logic

File: unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs [hunk 4]
```diff
@@ -240,31 +242,42 @@ public void TestAgent()
...
                 {
                     requestDecision +=1;
                     requestAction+=1;
                     agent2.RequestDecision();
                 }
-                else if (i % 5 == 0)
+                else if ((i % 5 == 0) && (i > 2))
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs]: comment this logic

File: unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs (hunk 2 above)
Comment (vincentpierre) [reply unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs]: Tabs converted

File: unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs (hunk 4 above)
Comment (vincentpierre) [reply unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs]: Comment is under this line

File: unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs [hunk 5]
```diff
@@ -185,16 +185,18 @@ public void TestAcademy()
...
                 Assert.AreEqual(1, aca.initializeAcademyCalls);
-                Assert.AreEqual(1, aca.episodeCount);
+                Assert.AreEqual(numberReset, aca.episodeCount);
                 Assert.AreEqual(i, aca.stepsSinceReset);
                 Assert.AreEqual(false, aca.IsDone());
-                Assert.AreEqual(1, aca.academyResetCalls);
+                Assert.AreEqual(numberReset, aca.academyResetCalls);
                 Assert.AreEqual(i, aca.AcademyStepCalls);
 
+                if (i == 0)
```
[the remark this answers is not in the document]
Comment (vincentpierre) [reply unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs]: Done

File: unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs [hunk 6]
```diff
@@ -62,7 +62,7 @@ public override void AgentOnDone()
 
     public class TestBrain : Brain
     {
-        // TODO : Mock a brain
+        // Mock a brain
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs]: Move comment to top of class with an explanation as to what purpose the class serves (hint: mock)

File: unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs [hunk 7]
```diff
@@ -185,16 +186,20 @@ public void TestAcademy()
...
-                Assert.AreEqual(1, aca.episodeCount);
+                Assert.AreEqual(numberReset, aca.episodeCount);
                 Assert.AreEqual(i, aca.stepsSinceReset);
                 Assert.AreEqual(false, aca.IsDone());
-                Assert.AreEqual(1, aca.academyResetCalls);
+                Assert.AreEqual(numberReset, aca.academyResetCalls);
                 Assert.AreEqual(i, aca.AcademyStepCalls);
 
+                if (i == 0)
+                { // The reset happens at the begining of the first step
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Editor/MLAgentsEditModeTest.cs]: Move comment above.

**Pull request merged**